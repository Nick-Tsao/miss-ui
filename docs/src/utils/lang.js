import Locale from '../../../packages/locale';
import zhCN from '../../../packages/locale/lang/zh-CN';
import enUS from '../../../packages/locale/lang/en-US';

const langMap = {
  'en-US': {
    title: 'Miss - Lightweight Mobile UI Components built on Vue',
    messages: enUS
  },
  'zh-CN': {
    title: 'Miss - 丰富、多样的移动端 Vue 组件库',
    messages: zhCN
  }
};
let currentLang = '';

function getDefaultLang() {
  const langs = Object.keys(langMap);
  const { hash } = location;

  for (let i = 0; i < langs.length; i++) {
    if (hash.indexOf(langs[i]) !== -1) {
      return langs[i];
    }
  }

  const userLang = localStorage.getItem('VANT_LANGUAGE') || navigator.language || 'en-US';
  return userLang.indexOf('zh-') !== -1 ? 'zh-CN' : 'en-US';
}

export function setLang(lang) {
  if (currentLang === lang) {
    return;
  }

  currentLang = lang;
  if (window.localStorage) {
    localStorage.setItem('VANT_LANGUAGE', lang);
  }
  Locale.use(lang, langMap[lang].messages);
  document.title = langMap[lang].title;
}

setLang(getDefaultLang());
