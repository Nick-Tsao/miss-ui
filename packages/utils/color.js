export const RED = '#f44';
export const BLUE = '#1989fa';
export const GREEN = '#4b0';
export const WHITE = '#fff';
export const GRAY_DARK = '#969799';
