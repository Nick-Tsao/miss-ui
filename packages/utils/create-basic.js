/**
 * Create a basic component with common options
 */
import '../locale';
import bem from '../mixins/bem';
import i18n from '../mixins/i18n';
import { isDef, camelize } from '.';

function install(Vue) {
  const { name } = this;
  Vue.component(name, this);
  Vue.component((camelize(`-${name}`)), this);
}

function returnArray() {
  return [];
}

function defaultProps(props) {
  Object.keys(props).forEach(key => {
    if (props[key] === Array) {
      props[key] = {
        type: Array,
        default: returnArray
      };
    } else if (props[key] === Number) {
      props[key] = {
        type: Number,
        default: 0
      };
    }
  });
}

export default function (sfc) {
  sfc.name = 'ms-' + sfc.name;
  sfc.install = sfc.install || install;
  sfc.mixins = sfc.mixins || [];
  sfc.mixins.push(i18n, bem);
  sfc.methods = sfc.methods || {};
  sfc.methods.isDef = isDef;
  sfc.props && defaultProps(sfc.props);

  return sfc;
}
