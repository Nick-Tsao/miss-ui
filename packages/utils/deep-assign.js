/* eslint-disable no-use-before-define */
import { isDef, isObj } from '.';

const { hasOwnProperty } = Object.prototype;

function assignKey(to, from, key) {
  const val = from[key];

  if (!isDef(val) || (hasOwnProperty.call(to, key) && !isDef(to[key]))) {
    return;
  }

  if (!hasOwnProperty.call(to, key) || !isObj(val)) {
    to[key] = val;
  } else {
    to[key] = assign(Object(to[key]), from[key]);
  }
}

export default function assign(to, from) {
  Object.keys(from).forEach(key => {
    assignKey(to, from, key);
  });

  return to;
}
