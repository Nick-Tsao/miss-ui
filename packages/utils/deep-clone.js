import deepAssign from './deep-assign';

export default function deepClone(obj) {
  if (Array.isArray(obj)) {
    return obj.map(item => deepClone(item));
  }
  if (typeof obj === 'object') {
    return deepAssign({}, obj);
  }
  return obj;
}
