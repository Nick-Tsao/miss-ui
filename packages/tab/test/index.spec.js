import Tab from '..';
import Tabs from '../../tabs';
import { mount, later, triggerDrag } from '../../../test/utils';

function createWrapper(options) {
  return mount({
    template: `
      <tabs @change="onChange"
        :color="color"
        :type="type"
        :swipeable="swipeable"
        :sticky="sticky"
        :line-width="lineWidth"
      >
        <tab :title="title1">Text</tab>
        <tab>
          <span slot="title">title2</span>
          Text
        </tab>
        <tab title="title3" disabled>Text</tab>
      </tabs>
    `,
    components: {
      Tab,
      Tabs
    },
    data() {
      return {
        title1: 'title1',
        color: '#f44',
        type: 'line',
        swipeable: true,
        sticky: true,
        lineWidth: 2
      };
    },
    ...options
  });
}

test('click to switch tab', async () => {
  const onChange = jest.fn();
  const wrapper = createWrapper({
    methods: {
      onChange
    }
  });

  await later();
  expect(wrapper).toMatchSnapshot();

  const tabs = wrapper.findAll('.ms-tab');
  tabs.at(1).trigger('click');
  tabs.at(2).trigger('click');
  await later();
  expect(wrapper).toMatchSnapshot();
  expect(onChange.mock.calls.length).toEqual(1);
});

test('swipe to switch tab', async () => {
  const onChange = jest.fn();
  const wrapper = createWrapper({
    methods: {
      onChange
    }
  });

  const content = wrapper.find('.ms-tabs__content');
  await later();
  expect(wrapper).toMatchSnapshot();
  triggerDrag(content, -100, 0);
  expect(wrapper).toMatchSnapshot();
  expect(onChange.mock.calls.length).toEqual(1);
  triggerDrag(content, -100, 0);
  expect(wrapper).toMatchSnapshot();
  triggerDrag(content, 100, 0);
  triggerDrag(content, 100, 0);
  expect(wrapper).toMatchSnapshot();

  await later();
  wrapper.destroy();
});

test('change tabs data', async () => {
  const wrapper = createWrapper();

  expect(wrapper).toMatchSnapshot();

  wrapper.setData({
    swipeable: false,
    sticky: false,
    type: 'card',
    color: 'blue',
    title1: 'new title1'
  });
  expect(wrapper).toMatchSnapshot();
});
