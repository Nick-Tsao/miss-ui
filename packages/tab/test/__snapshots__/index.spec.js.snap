// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change tabs data 1`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line"></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <div><span>title2</span></div>
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`change tabs data 2`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line"></div>
      <div class="ms-tab ms-tab--active"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab"><span class="ms-ellipsis">

        </span></div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="">
      <!---->
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <div><span>title2</span></div>
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`click to switch tab 1`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line" style="width: 2px; background-color: rgb(255, 68, 68); transform: translateX(-1px);"></div>
      <div class="ms-tab ms-tab--active"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab">
        <div><span>title2</span></div>
      </div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="">Text
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`click to switch tab 2`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line" style="width: 2px; background-color: rgb(255, 68, 68); transform: translateX(-1px);"></div>
      <div class="ms-tab"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab ms-tab--active">
        <div><span>title2</span></div>
      </div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="display: none;">Text
      <!---->
    </div>
    <div class="ms-tab__pane" style="">
      Text
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 1`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line" style="width: 2px; background-color: rgb(255, 68, 68); transform: translateX(-1px);"></div>
      <div class="ms-tab ms-tab--active"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab">
        <div><span>title2</span></div>
      </div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="">Text
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 2`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line" style="width: 2px; background-color: rgb(255, 68, 68); transform: translateX(-1px);"></div>
      <div class="ms-tab"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab ms-tab--active">
        <div><span>title2</span></div>
      </div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="display: none;">Text
      <!---->
    </div>
    <div class="ms-tab__pane" style="">
      Text
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 3`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line" style="width: 2px; background-color: rgb(255, 68, 68); transform: translateX(-1px);"></div>
      <div class="ms-tab"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab ms-tab--active">
        <div><span>title2</span></div>
      </div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="display: none;">Text
      <!---->
    </div>
    <div class="ms-tab__pane" style="">
      Text
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`swipe to switch tab 4`] = `
<div class="ms-tabs ms-tabs--line">
  <div class="ms-tabs__wrap ms-hairline--top-bottom">
    <div class="ms-tabs__nav ms-tabs__nav--line" style="border-color: #f44;">
      <div class="ms-tabs__line" style="width: 2px; background-color: rgb(255, 68, 68); transform: translateX(-1px);"></div>
      <div class="ms-tab ms-tab--active"><span class="ms-ellipsis">
          title1
        </span></div>
      <div class="ms-tab">
        <div><span>title2</span></div>
      </div>
      <div class="ms-tab ms-tab--disabled"><span class="ms-ellipsis">
          title3
        </span></div>
    </div>
  </div>
  <div class="ms-tabs__content">
    <div class="ms-tab__pane" style="">Text
      <!---->
    </div>
    <div class="ms-tab__pane" style="display: none;">
      Text
    </div>
    <div class="ms-tab__pane" style="display: none;">
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
